crates/ty_python_semantic/src/site_packages.rs

```diff
@@ -933,6 +1042,92 @@ fn site_packages_directory_from_sys_prefix(
     ))
 }
 
+/// Attempt to retrieve the real stdlib directory
+/// associated with a given Python installation.
+///
+/// The location of the stdlib directory can vary according to the
+/// Python version that this installation represents. The Python version may
+/// or may not be known at this point, which is why the `python_version`
+/// parameter is an `Option`.
+fn real_stdlib_directory_from_sys_prefix(
+    sys_prefix_path: &SysPrefixPath,
+    python_version: Option<PythonVersion>,
+    implementation: PythonImplementation,
+    system: &dyn System,
+) -> StdlibDiscoveryResult<SystemPathBuf> {
+    tracing::debug!(
+        "Searching for real stdlib directory in sys.prefix {}",
+        sys_prefix_path.inner
+    );
+
+    if cfg!(target_os = "windows") {
+        let stdlib = sys_prefix_path.join("Lib");
+        return system.is_directory(&stdlib).then_some(stdlib).ok_or(
+            StdlibDiscoveryError::NoStdlibFound(sys_prefix_path.to_owned()),
+        );
+    }
+
+    // If we were able to figure out what Python version this installation is,
+    // we should be able to avoid iterating through all items in the `lib/` directory:
+    if let Some(expected_relative_path) = implementation.relative_stdlib_path(python_version) {
+        let expected_absolute_path = sys_prefix_path.join(expected_relative_path);
+        if system.is_directory(&expected_absolute_path) {
+            return Ok(expected_absolute_path);
+        }
+
+        // CPython free-threaded (3.13+) variant: pythonXYt
+        if matches!(implementation, PythonImplementation::CPython)
+            && python_version.is_some_and(PythonVersion::free_threaded_build_available)
+        {
+            let alternative_path =
+                sys_prefix_path.join(format!("lib/python{}t", python_version.unwrap()));
+            if system.is_directory(&alternative_path) {
+                return Ok(alternative_path);
+            }
+        }
+    }
```

Comment:
Nit: If a small preference to use a `match` or `let` here for `python_version` to avoid the unwrap. It's, unfortunately, a bit more verbose without if let chains (only one more month before we can use them)


```suggestion
        match (implementation, python_version) {
            (PythonImplementation::CPython, Some(python_version)) if python_version.free_threaded_build_available() => {
                let alternative_path =
                sys_prefix_path.join(format!("lib/python{}t", python_version.unwrap()));
                if system.is_directory(&alternative_path) {
                    return Ok(alternative_path);
                }
            }
        }
        if matches!(implementation, PythonImplementation::CPython)
            && python_version.is_some_and(PythonVersion::free_threaded_build_available)
        {
            
        }
    }
```

Reply:
note that this is copy-pasted logic from the equivalent site-packages logic